libs/model/src/models/contest_action.ts

```diff
@@ -0,0 +1,50 @@
+import { schemas } from '@hicommonwealth/core';
+import Sequelize from 'sequelize';
+import { z } from 'zod';
+import type { ModelInstance, ModelStatic } from './types';
+
+type ContestAction = ModelInstance<
+  z.infer<typeof schemas.projections.ContestAction>
+>;
+
+export default (sequelize: Sequelize.Sequelize) =>
+  <ModelStatic<ContestAction>>sequelize.define<ContestAction>(
+    'ContestAction',
+    {
+      contestAddress: {
```

Comment:
What are the advantages of a (5 column!) composite primary key over a surrogate key here? Are we 1 million percent sure we will never need to reference this table? Also, we would expect there to be _a lot_ of contest actions so wouldn't it be more performant for joins to have a surrogate key and then additional indexes as required for performance? We need to review the kinds of queries we will need to make ahead of time if we want to use a composite primary key like this.

Reply:
this is a read-only table, the projection is the only (async) writer, so we have to optimize for reading. Since we don't have all the query requirements defined yet, this is kind of a general key, but we will tweak this view for optimal query performance once those requirements are finished.